src/build.js

```diff
@@ -1,38 +1,142 @@
 import path from 'path';
 import Zip from 'jszip';
+import Promise, { all, promisifyAll, resolve } from 'bluebird';
 import {
     __,
+    complement,
     concat,
     contains,
     curry,
+    drop,
+    equals,
     filter,
+    identity,
+    join,
+    lensProp,
     map,
+    mapObjIndexed,
+    mergeAll,
+    over,
+    pick,
     pipe,
+    prop,
     sort,
     subtract,
+    takeWhile,
     test,
+    tryCatch,
     type,
     without
 } from 'ramda';
-import Promise, { promisifyAll } from 'bluebird';
+import { emitWarning } from './input';
+import { compileIndex } from './run';
+import { getProperties } from './vm';
 
 const fs = promisifyAll(require('fs'));
 
 const defaultFileOptions = { date: new Date(1149562800000) };
-
 const requiredFiles = ['package.json', 'index.js'];
-const projectFiles = concat(requiredFiles, ['icon.png', 'yarn.lock']);
+const projectFiles = ['icon.png', ...requiredFiles];
+
+const localeByFile = pipe(
+    drop(8),
+    takeWhile(complement(equals('.'))),
+    join('')
+);
+
+/**
+ * Converts a list of locale files to pairs containing locale strng and content
+ *
+ * @param {String[]} localeFiles
+ * @return {Promise}
+ */
+function localesToPairs(localeFiles) {
+    return all(localeFiles.map(localeFile => fs.readFileAsync(localeFile, 'utf-8')
+        .then(JSON.parse)
+        .then(json => [localeByFile(localeFile), json])));
+}
+
+/**
+ * Transform the config fields, extracting what should be uploaded and getting
+ * only the description from parameters
+ *
+ * @param {Object} *
+ * @return {Object}
+ */
+const transformConfigFields = pipe(
+    pick(['title', 'description', 'preview', 'params']),
+    over(lensProp('params'), mapObjIndexed(pick(['description', 'default']))));
+
+/**
+ * Lazily runs the extension using all possible listed locales and extracts
+ * the meta-data
+ *
+ * @param {[(String, *)]} locales
+ * @param {String} source
+ * @return {Promise}
+ */
+function runInAllLocales(locales, source) {
+    return all([['default', {}], ...locales].map(([locale, strings]) =>
+        getProperties({ name: `precompile-${locale}`, source }, strings)
+            .then(config => ({ [locale]: transformConfigFields(config) }))))
+            .then(mergeAll);
+}
 
-function assertRequiredFiles(files) {
+/**
+ * Creates a meta file where the information about precompilation is stored
+ *
+ * @param {Object} locales
+ * @return {Promise}
+ */
+function createMetaFile(locales) {
+    return fs.writeFileAsync('.meta', JSON.stringify(locales));
+}
+
+/**
+ * Precompiles the locale files, generating a meta file containing the meta
+ *
+ * @param {String[]} files
+ * @return {Promise}
+ */
+function precompileLocales(files) {
+    return resolve(files)
+        .then(filter(test(/^locales\/[a-z]{2}(_[A-Z]{2,3})?\.json$/)))
```

Comment:
But the regex is correct. It matches against `pt` and `pt_BR`, there is no need to call an external process, Unix-only for that. There is a case for the language characters and an optional case for the dialect.